STUDIO/org.openl.rules.webstudio/src/org/openl/rules/webstudio/security/LdapToOpenLUserDetailsMapper.java

```diff
@@ -68,6 +69,27 @@ public UserDetails mapUserFromContext(DirContextOperations ctx,
 
         String firstName = ctx.getStringAttribute("givenName");
         String lastName = ctx.getStringAttribute("sn");
+        String email = ctx.getStringAttribute("mail");
+        String displayName = ctx.getStringAttribute("displayName");
+        boolean firstNameExternal = false;
+        boolean lastNameExternal = false;
+        boolean emailExternal = false;
+        boolean displayNameExternal = false;
+        if (StringUtils.isBlank(displayName)) {
+            displayName = ctx.getStringAttribute("cn");
+        }
+        if (StringUtils.isNotBlank(firstName)) {
+            firstNameExternal = true;
+        }
+        if (StringUtils.isNotBlank(lastName)) {
+            lastNameExternal = true;
+        }
+        if (StringUtils.isNotBlank(email)) {
+            emailExternal = true;
+        }
+        if (StringUtils.isNotBlank(displayName)) {
+            displayNameExternal = true;
+        }

```

Comment:
Can be simplified like:
```suggestion
        if (StringUtils.isBlank(displayName)) {
            displayName = ctx.getStringAttribute("cn");
        }
        boolean firstNameExternal = StringUtils.isNotBlank(firstName);
        boolean lastNameExternal = StringUtils.isNotBlank(lastName);
        boolean emailExternal = StringUtils.isNotBlank(email);
        boolean displayNameExternal = StringUtils.isNotBlank(displayName);
```

Or even it can be replaced with int value, for example, by default all bits in int are zero, but if ``firstNameExternal`` flag is enabled, then set first bit to ``1``, if lastNameExternal is enabled then set the second bit to ``1`` and etc.
Setting and checking of ane feature can be done using mask like:

Enable any flag example:
```
userExternalFlags |= Flags.FIRST_NAME_EXTERNAL.getMask();
```

Disable any flag example:
```
userExternalFlags &= ~Flags.FIRST_NAME_EXTERNAL.getMask();
```

Check if any flag enabled example:
```
(userExternalFlags & Flags.FIRST_NAME_EXTERNAL.getMask()) != 0
```

Reply:
Fixed.